webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,660 @@
+    }
+    return array;
+  }
+
+  private String getMachine(HttpServletRequest request) {
+    SettingBundle settings =
+        ResourceLocator.getSettingBundle("org.silverpeas.webSites.settings.webSiteSettings");
+    SettingBundle generalSettings = ResourceLocator.getGeneralSettingBundle();
+
+    StringBuilder machine = new StringBuilder(settings.getString("Machine", "").trim());
+    String context = (generalSettings.getString("ApplicationURL")).substring(1);
+    if (machine.length() == 0) {
+      List<String> a = buildTab(request.getRequestURL().toString());
+      int j = 1;
+      while (true) {
+        if (j > a.size()) {
+          break;
+        }
+
+        if (!a.get(j).equals(context)) {
+          if (machine.length() == 0) {
+            machine.append(a.get(j));
+          } else {
+            machine.append("/").append(a.get(j));
+          }
+        } else {
+          break;
+        }
+        j++;
+      }
+    }
+    return machine.toString();
+  }
+
+  public String ignoreAntiSlash(String chemin) {
+    /* ex : \\\rep1\\rep2\\rep3 */
+    /* res = rep1\\rep2\\re3 */
+
+    String res = chemin;
+    boolean ok = false;
+    while (!ok) {
+      char car = res.charAt(0);
+      if (car == '\\') {
+        res = res.substring(1);
+      } else {
+        ok = true;
+      }
+    }
+    return res;
+
+  }
+
+  public String removeDoubleAntiSlash(String chemin) {
+    /* ex : id\\rep1\\rep11\\rep111 */
+    /* res = id\rep1\rep11\re111 */
+    return chemin.replaceAll("\\\\\\\\", "\\\\");
+  }
+
+  public String finNode(WebSiteSessionController scc, String path) {
+    /* ex : ....webSite17\\id\\rep1\\rep2\\rep3 */
+    /* res : id\rep1\rep2\rep3 */
+
+    int longueur = scc.getComponentId().length();
+    int index = path.lastIndexOf(scc.getComponentId());
+    String chemin = path.substring(index + longueur);
+
+    chemin = ignoreAntiSlash(chemin);
+    chemin = removeDoubleAntiSlash(chemin);
+
+    return chemin;
+  }
+
+  private List<String> sortCommun(List<String> tabContexte, List<String> tab) {
+    /* tabContexte = [id | rep1 | rep2] */
+    /* tab = [id | rep1 | rep3] */
+    /* res = [id | rep1] */
+    int i = 0;
+    boolean ok = true;
+    List<String> array = new ArrayList<String>();
+
+    while (ok && i < tabContexte.size()) {
+      String contenuContexte = tabContexte.get(i);
+      if (i < tab.size()) {
+        String contenu = tab.get(i);
+        if (contenuContexte.equals(contenu)) {
+          array.add(contenu);
+
+        } else {
+          ok = false;
+        }
+        i++;
+      } else {
+        ok = false;
+      }
+    }
+    return array;
+  }
+
+  private String sortReste(List<String> tab, List<String> tabCommun) {
+    /* tab = [id | rep1 | rep2 | rep3] */
+    /* tabCommun = [id | rep1] */
+    /* res = rep2/rep3 */
+    String res = "";
+
+    int indice = tabCommun.size();
+
+    while (indice < tab.size()) {
+      String contenu = tab.get(indice);
+      res += contenu + "/";
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) org.silverpeas.components.websites.servlets.design.SiteDesignActionHandler.sortReste(List, List) concatenates strings using + in a loop [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=findbugs%3ASBSC_USE_STRINGBUFFER_CONCATENATION)